protocol-units/bridge/chains/ethereum/src/lib.rs

```diff
@@ -0,0 +1,324 @@
+use alloy_network::{Ethereum, EthereumWallet};
+use alloy_primitives::private::serde::{Deserialize, Serialize};
+use alloy_primitives::{FixedBytes, U256};
+use alloy_provider::{
+	fillers::{ChainIdFiller, FillProvider, GasFiller, JoinFill, NonceFiller, WalletFiller},
+	Provider, ProviderBuilder, RootProvider,
+};
+use alloy_rlp::{Decodable, Encodable, RlpDecodable, RlpEncodable};
+use alloy_sol_types::sol;
+use alloy_transport::BoxTransport;
+use alloy_transport_ws::WsConnect;
+use anyhow::Context;
+use bridge_shared::types::{
+	Amount, BridgeTransferDetails, BridgeTransferId, CounterpartyCompletedDetails, HashLock,
+	HashLockPreImage, InitiatorAddress, RecipientAddress, TimeLock,
+};
+use bridge_shared::{
+	bridge_contracts::{
+		BridgeContractCounterpartyError, BridgeContractInitiator, BridgeContractInitiatorError,
+		BridgeContractInitiatorResult,
+	},
+	bridge_monitoring::{BridgeContractCounterpartyEvent, BridgeContractInitiatorEvent},
+	types::LockDetails,
+};
+use keccak_hash::keccak;
+use mcr_settlement_client::send_eth_transaction::{
+	send_transaction, InsufficentFunds, SendTransactionErrorRule, UnderPriced, VerifyRule,
+};
+use std::fmt::Debug;
+use thiserror::Error;
+use utils::EthAddress;
+
+pub mod utils;
+
+const INITIATOR_ADDRESS: &str = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
+const COUNTERPARTY_ADDRESS: &str = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"; //Dummy val
+const RECIPIENT_ADDRESS: &str = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
+const DEFAULT_GAS_LIMIT: u64 = 10_000_000_000;
+const MAX_RETRIES: u32 = 5;
+
+type EthHash = [u8; 32];
+
+pub type SCIResult<A, H> = Result<BridgeContractInitiatorEvent<A, H>, BridgeContractInitiatorError>;
+pub type SCCResult<A, H> =
+	Result<BridgeContractCounterpartyEvent<A, H>, BridgeContractCounterpartyError>;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum MoveCounterpartyEvent<A, H> {
+	LockedBridgeTransfer(LockDetails<A, H>),
+	CompletedBridgeTransfer(CounterpartyCompletedDetails<A, H>),
+}
+
+#[derive(Debug, Error, Clone, PartialEq, Eq)]
+pub enum MoveCounterpartyError {
+	#[error("Transfer not found")]
+	TransferNotFound,
+	#[error("Invalid hash lock pre image (secret)")]
+	InvalidHashLockPreImage,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum EthInitiatorEvent<A, H> {
+	InitiatedBridgeTransfer(BridgeTransferDetails<A, H>),
+	CompletedBridgeTransfer(BridgeTransferId<H>, HashLockPreImage),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AbstractBlockainEvent<A, H> {
+	InitiatorContractEvent(SCIResult<A, H>),
+	CounterpartyContractEvent(SCCResult<A, H>),
+	Noop,
+}
+
+///Configuration for the Ethereum Bridge Client
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Config {
+	pub rpc_url: Option<String>,
+	pub ws_url: Option<String>,
+	pub chain_id: String,
+	pub signer_private_key: String,
+	pub initiator_address: String,
+	pub counterparty_address: String,
+	pub recipient_address: String,
+	pub gas_limit: u64,
+	pub num_tx_send_retries: u32,
```

Comment:
The other Config types have been update but `signer_private_key` field seems better as a `String` because the type it parses to `LocalSigner<SigningKey<Secp256k1>>` is not `Deserialize`. And `Config needs to be deserializable if we want to do:
```
fn load_config(path: &str) -> Result<Config> {
	match std::fs::read_to_string(path) {
		Ok(config_str) => {
			let config: Config = serde_json::from_str(&config_str)?;
			Ok(config)
		}
		Err(_) => {
			println!("Config file not found, Using default config values");
			Ok(Config::default())
		}
	}
}
```
Which allows a config to be loaded from a path.

Reply:
The fields on the client have been updated to their native chain types. However we still need the fields as `send_transaction` requires them 

```rust
let _ = send_transaction(
			call,
			&self.send_transaction_error_rules,
			self.num_tx_send_retries,
			self.gas_limit as u128,
		)
```
The values never get changed. So if we don't want them on the client they would need to be stored somewhere else in memory, and then passed in. For now  I think its easy and ergonomic enough for them to remain on the client. Maybe after we're done we will see somewhere where they can be accessed.